go-app-ussd_pmtct_seed.js

```diff
@@ -113,7 +192,19 @@ go.app = function() {
 
         self.add("state_start", function(name) {
             self.im.user.answers = {};  // reset answers
-            return self.states.create("state_check_pmtct_subscription");
+            var address = utils.normalize_msisdn(self.im.user.addr, '27');
+            // NOTE:    We're making the API call here but not telling it to wait nor are we doing
+            //          anything with the result.
+            //
+            //          The idea is that the check continues to happen in the background and will be ready
+            //          when we need it. This way we minimise any timeout penalty during the registration.
+            return self
+                .is_valid_recipient_for_pilot({
+                    address: address,
+                    wait: false
+                }).then(function(res) {
+                    return self.states.create("state_check_pmtct_subscription");
```

Comment:
Do we need to check the return value of `.is_valid_recipient_for_pilot()` or does that happen automatically somehow?

Reply:
The return value of `is_valid_recipient_for_pilot` is sent as the argument to the function specified in `.then`, which in this case is `res`.

But in this case we ignore it. This is the `wait: false` call, which just tells the API to do and cache the query, so that when we call it later with `wait: true`, it has hopefully by that point completed and we get a quick response.

If you're talking about checking the response code, then yes, that should go through the errback, which by default should trigger the error state.

Reply:
Makes sense, thanks! 👍